pkg/db/src/implementations/version-1.1/note.ts

```diff
@@ -250,11 +256,19 @@ class Note implements NoteInterface {
 			warehouseId: warehouseId ? versionId(warehouseId) : this.noteType === "inbound" ? this.#w._id : ""
 		};
 
-		const i = entries.findIndex((e) => e.isbn === transaction.isbn && e.warehouseId === transaction.warehouseId);
+		const updateTransaction = {
+			isbn: update.isbn,
+			quantity: update.quantity,
+			warehouseId: update.warehouseId ? versionId(update.warehouseId) : this.noteType === "inbound" ? this.#w._id : ""
+		};
+
+		const i = entries.findIndex(
+			(e) => e.isbn === transaction.isbn && e.warehouseId === transaction.warehouseId && e.quantity === transaction.quantity
+		);
```

Comment:
Quantity shouldn't be used to match the transaction...it's completely unnecessary